src/charm.py

```diff
@@ -474,7 +477,7 @@ def _on_peer_relation_changed(self, event: HookEvent) -> None:
             event.defer()
             return
         else:
-            self.unit_peer_data.pop("start-tls.server", None)
+            self.unit_peer_data.pop("start-tls-server", None)
```

Comment:
This is an interesting catch... 

Q: why do we set/remove it if we do not check it?
```
> ripgrep start-tls.server
src/charm.py
470:            self.unit_peer_data.update({"start-tls-server": "True"})
477:            self.unit_peer_data.pop("start-tls.server", None)
```

Reply:
I'm unsure, so I left it be for the time being.